api/src/data_migration/load/load_oracle_data_task.py

```diff
@@ -0,0 +1,176 @@
+
+        self.log_row_count("row count after", foreign_table, staging_table)
+
+    def do_insert(self, foreign_table: sqlalchemy.Table, staging_table: sqlalchemy.Table) -> int:
+        """Determine new rows by primary key, and copy them into the staging table."""
+
+        insert_from_select_sql, select_sql = sql.build_insert_select_sql(
+            foreign_table, staging_table
+        )
+
+        # print(insert_from_select_sql)
+
+        # COUNT has to be a separate query as INSERTs don't return a rowcount.
+        insert_count = self.db_session.query(select_sql.subquery()).count()
+
+        self.increment("count.insert.total", insert_count)
+        self.set_metrics({f"count.insert.{staging_table.name}": insert_count})
+
+        # Execute the INSERT.
+        t0 = time.monotonic()
+        self.db_session.execute(insert_from_select_sql)
+        t1 = time.monotonic()
+
+        self.set_metrics({f"time.insert.{staging_table.name}": round(t1 - t0, 3)})
+
+        return insert_count
+
+    def do_update(self, foreign_table: sqlalchemy.Table, staging_table: sqlalchemy.Table) -> int:
+        """Find updated rows using last_upd_date, copy them, and reset transformed_at to NULL."""
+
+        update_sql = sql.build_update_sql(foreign_table, staging_table).values(transformed_at=None)
+
+        # print(update_sql)
+        t0 = time.monotonic()
+        result = self.db_session.execute(update_sql)
+        t1 = time.monotonic()
+        update_count = result.rowcount
+
+        self.increment("count.update.total", update_count)
+        self.set_metrics({f"count.update.{staging_table.name}": update_count})
+        self.set_metrics({f"time.update.{staging_table.name}": round(t1 - t0, 3)})
+
+        return update_count
+
+    def do_mark_deleted(
+        self, foreign_table: sqlalchemy.Table, staging_table: sqlalchemy.Table
+    ) -> int:
+        """Find deleted rows, set is_deleted=TRUE, and reset transformed_at to NULL."""
+
+        update_sql = sql.build_mark_deleted_sql(foreign_table, staging_table).values(
+            transformed_at=None,
+            deleted_at=datetime_util.utcnow(),
+        )
+
+        # print(update_sql)
+        t0 = time.monotonic()
+        result = self.db_session.execute(update_sql)
+        t1 = time.monotonic()
+        delete_count = result.rowcount
+
+        self.increment("count.delete.total", delete_count)
+        self.set_metrics({f"count.delete.{staging_table.name}": delete_count})
+        self.set_metrics({f"time.delete.{staging_table.name}": round(t1 - t0, 3)})
+
+        return delete_count
+
+    def log_row_count(self, message: str, *tables: sqlalchemy.Table) -> None:
+        """Log the number of rows in each of the tables using SQL COUNT()."""
+        extra = {}
+        for table in tables:
+            extra[f"count.{table.schema}.{table.name}"] = self.db_session.query(table).count()
+        logger.info(message, extra=extra, stacklevel=2)
```

Comment:
Could we add these to the metrics object? Probably as something like `before.count.{schema}.{name}` and `after.count...`. That way the counts are all in one log message at the end of the run which will make aggregating them in New Relic simpler.